underfs/local/src/main/java/alluxio/underfs/local/LocalUnderFileSystem.java

```diff
@@ -289,6 +291,21 @@ public void connectFromWorker(String hostname) throws IOException {
   }
 
   /**
+   * Rename a file to a file or a directory to a directory.
+   *
+   * @param src path of source file or directory
+   * @param dst path of destination file or directory
+   * @return true if rename succeeds
+   * @throws IOException
+   */
+  private boolean rename(String src, String dst) throws IOException {
```

Comment:
This internal rename is called my renameFile and renameDirectory. There are no extra checks in the those functions in the present implementation => the caveat being that for LocalUFS, renameFile will succeed even when operating on a directory (and renameDirectory on file).